src/main/java/io/mathdojo/useraccountservice/services/IdentityService.java

```diff
@@ -164,16 +159,13 @@ public User updateUserWithId(String orgId, String userId, AccountRequest account
         boolean verificationStatusToUpdate = (!foundUser.isAccountVerified()
                 && accountModificationRequest.isAccountVerified()) ? accountModificationRequest.isAccountVerified()
                         : foundUser.isAccountVerified();
-        return new User(userId, verificationStatusToUpdate, nameToUpdate, profileToUpdate, returnedOrgId);
+        return userRepo.save(new User(userId, verificationStatusToUpdate, nameToUpdate, profileToUpdate, returnedOrgId));
     }
 
 	public void deleteUserFromOrg(String orgId, String userId) {
-        String returnedOrgId = (getOrganisationById(orgId)).getId();
-        if (PRECONDITIONED_UNKNOWN_USER_ID.equals(userId)) {
-            targetExecutionContext.getLogger().log(Level.WARNING,
-                    String.format("UserId %s in Org %s could not be found", userId, orgId));
-            throw new IdentityServiceException(UNKNOWN_USERID_EXCEPTION_MSG);
-        }
+        User retrievedUser = getUserInOrg(orgId, userId);
+        userRepo.delete(retrievedUser);
+        
```

Comment:
Although the scope of #27 was allowing existing users to be retrieved i￼￼t￼ seems we're now doing a bigger piece of updating all user and org management functionality to integrate with the repository? Have I understood that right?

Reply:
Just the user functionality. Once the preconditioned users were removed it was really just a 3 line change to add the other functionality(162, 166 and 167) and it was a convenient way to see that the test changes worked.